File: app/src/main/java/d/spidchenko/canvasgame/MainActivity.java
package d.spidchenko.canvasgame;

import android.graphics.Point;
import android.os.Bundle;
import android.util.Log;
import android.view.MotionEvent;
import android.view.View;
import android.widget.LinearLayout;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity implements View.OnTouchListener {

    private static final String TAG = "MainActivity.LOG_TAG";
    public static Point tapPosition;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        GameView gameView = new GameView(this); // создаём gameView

        LinearLayout gameLayout = (LinearLayout) findViewById(R.id.gameLayout); // находим gameLayout
        gameLayout.addView(gameView); // и добавляем в него gameView

        gameLayout.setOnTouchListener(this);

    }

    @Override
    public boolean onTouch(View v, MotionEvent event) {
        Log.d(TAG, "onTouch: " + (int) event.getX() + " " + (int) event.getY());
        tapPosition = new Point((int) event.getX(), (int) event.getY());
        return false;
    }
}

File: app/src/main/java/d/spidchenko/canvasgame/MainActivity.kt
package d.spidchenko.canvasgame

import android.app.Activity
import android.graphics.Point
import android.os.Bundle
import android.view.MotionEvent
import android.view.View
import android.view.Window
import android.widget.LinearLayout

class MainActivity : Activity(), View.OnTouchListener, View.OnClickListener,
    View.OnLongClickListener {

    lateinit var gameEngine: GameEngine

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        requestWindowFeature(Window.FEATURE_NO_TITLE)
        setContentView(R.layout.activity_main)
        gameEngine = GameEngine(this)
        val gameLayout: LinearLayout = findViewById(R.id.gameLayout)
        gameLayout.addView(gameEngine)
        gameLayout.setOnTouchListener(this)
        gameLayout.setOnClickListener(this)
        gameLayout.setOnLongClickListener(this)
    }

    override fun onLongClick(p0: View?): Boolean {
        clickDuration = ClickDuration.LONG
        return true
    }

    override fun onClick(p0: View?) {
        clickDuration = ClickDuration.SHORT
    }

    override fun onTouch(v: View, event: MotionEvent): Boolean {
        if (event.action == MotionEvent.ACTION_UP) {
            lastClickCoordinates = Point(event.x.toInt(), event.y.toInt())
        }
        return false
    }

    override fun onResume() {
        super.onResume()
        gameEngine.resume()
    }

    override fun onPause() {
        super.onPause()
        gameEngine.pause()
    }

    companion object {
        private const val TAG = "MainActivity.LOG_TAG"

        // TODO need better logic for handling input events
        var lastClickCoordinates: Point? = null
        var clickDuration = ClickDuration.SHORT
    }
}